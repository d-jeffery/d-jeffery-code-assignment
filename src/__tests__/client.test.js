const { Client } = require('../client');
const { Provider } = require('../provider');

describe('Client', () => {
  test('create with id', () => {
    const client = new Client('Mr Harrison');

    expect(client.id).toEqual('Mr Harrison');
  });

  test('empty bookings should have 0 entries', () => {
    const client = new Client('Mr Harrison');

    expect(client.bookings.size).toBe(0);
  });

  test('add booking should result in 1 entry', () => {
    const client = new Client('Mr Harrison');
    const provider = new Provider('Dr Jones');
    client.addBooking(new Date(1723680000000).getTime(), provider.id);

    expect(client.bookings.size).toBe(1);
    expect(client.getBookings()).toMatchSnapshot();
  });
});
