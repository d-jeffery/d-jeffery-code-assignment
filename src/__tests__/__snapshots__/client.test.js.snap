// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client add booking should result in 1 entry 1`] = `
Map {
  1723680000000 => "Mr Harrison",
}
`;
